internal/controller/siteconfig_helper.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"reflect"
+	"strings"
+
+	sprig "github.com/go-task/slim-sprig"
+	"github.com/sakhoury/siteconfig/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+type SpecialVars struct {
+	CurrentNode                      v1alpha1.NodeSpec
+	InstallConfigOverrides           string
+	ControlPlaneAgents, WorkerAgents int
+}
+
+// SiteData is a special object that provides an interface to the SiteConfig spec fields for use in rendering templates
+type SiteData struct {
+	Site        v1alpha1.SiteConfigSpec
+	SpecialVars SpecialVars
+}
+
+// getConfigMap retrieves the configmap from cluster for a given TemplateRef
+func getConfigMap(ctx context.Context, c client.Client, templateRef v1alpha1.TemplateRef) (*corev1.ConfigMap, error) {
```

Comment:
I don't think you need a function for this.
It's just an alias since it's a single function call.

Reply:
removed